typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -754,7 +754,19 @@
                                                                      "property"
                                                                      #,real-prop-var)
                                               (#,pred? x)))))]
-       [(Prefab: key (list (app t->sc fld/scs) ...)) (prefab/sc key fld/scs)]
+       [(Prefab: (and key (list key-sym rst ...)) (list fld/scs^ ...))
```

Comment:
Can we abstract out some of this code, since it's basically the same as for `struct`?